embedded-graphics/src/image/mod.rs

```diff
@@ -1,33 +1,262 @@
+//! Image support for embedded-graphics
 //!
-//! Image drawables can be created for raw bitmap data and images in BMP and TGA
-//! format.
+//! Adding embedded-graphics support to an image format requires the implementation of the
+//! [`ImageDimensions`] and [`IntoPixelIter`] traits. These provide a common interface to image metadata
+//! and an iterator over individual pixel values respectively.
+//!
+//! The [`Image`] struct is a wrapper around items that implement [`ImageDimensions`] and allows them to
+//! be drawn to a [`DrawTarget`], reading pixel values from the implementation of
+//! [`IntoPixelIter`].
+//!
+//! # Examples
+//!
+//! ## Load a TGA image and draw it to a display
+//!
+//! This example loads a TGA-formatted image using the [tinytga] crate and draws it to the display
+//! using the [`Image`] wrapper. The image is positioned at the top left corner of the display.
+//!
+//! ```rust
+//! use embedded_graphics::{image::Image, pixelcolor::Rgb565, prelude::*};
+//! # use embedded_graphics::mock_display::MockDisplay as Display;
+//! use tinytga::Tga;
+//!
+//! let mut display: Display<Rgb565> = Display::default();
+//!
+//! let tga =
+//!     Tga::from_slice(include_bytes!("../../../simulator/examples/rust-pride.tga")).unwrap();
+//!
+//! let image = Image::new(&tga, Point::zero());
+//!
+//! image.draw(&mut display);
+//!
+//! # Ok::<(), core::convert::Infallible>(())
+//! ```
+//!
+//! [tinytga]: https://crates.io/crates/tinytga
+//! [`IntoPixelIter`]: ./trait.IntoPixelIter.html
+//! [`ImageDimensions`]: ./trait.ImageDimensions.html
+//! [`Image`]: ./struct.Image.html
+//! [`DrawTarget`]: ../trait.DrawTarget.html
 
-#[cfg(feature = "bmp")]
-mod image_bmp;
 mod image_raw;
-#[cfg(feature = "tga")]
-mod image_tga;
 
-pub use self::image_raw::{Image, ImageBE, ImageLE};
+pub use self::image_raw::{ImageRaw, ImageRawBE, ImageRawLE};
 
-#[cfg(feature = "bmp")]
-pub use self::image_bmp::ImageBmp;
-#[cfg(feature = "tga")]
-pub use self::image_tga::ImageTga;
+use crate::{
+    draw_target::DrawTarget,
+    drawable::{Drawable, Pixel},
+    geometry::{Dimensions, Point, Size},
+    pixelcolor::PixelColor,
+    transform::Transform,
+};
+use core::{fmt, marker::PhantomData};
 
-/// Image file trait.
-pub trait ImageFile<'a>: crate::geometry::Dimensions + Sized {
-    /// Create a new image with given input file
-    ///
-    /// The input file is expected to be of a particular format (BMP, TGA, etc) and contain file
-    /// metadata like width/height and pixel data. Because parsing may fail, this returns a
-    /// `Result<Self, ()>`.
-    fn new(filedata: &'a [u8]) -> Result<Self, ()>;
+/// Image data iterator trait
+pub trait IntoPixelIter<C>
+where
+    C: PixelColor + From<<C as PixelColor>::Raw>,
+{
+    /// Iterator over pixels in the image
+    type PixelIterator: Iterator<Item = Pixel<C>>;
+
+    /// Get an iterator over the pixels of the image
+    fn pixel_iter(self) -> Self::PixelIterator;
+}
 
+/// Image data trait
```

Comment:
There doesn't seem to be one default way to write these summaries for trait docs (At least the `std` docs uses different ways). I would like them to be somewhat consistent across e-g. This would by one variant:
```suggestion
/// A trait to get the dimensions of an image.
```
What is your preferred way of writing these?